### PR TITLE
Adding endpoints to save & write Fingerprinting Results

### PR DESCRIPTION
**Pull request by DavidStreid**
Fingerprinting results are stored in a crosscheck_metrics file. This is parsed and saved to a database that can be queried

### CODE REVIEW

File: src/main/java/org/mskcc/constants.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.mskcc;
+
+public class constants {
```
Comment (dmcmanam) [inline src/main/java/org/mskcc/constants.java]: public class Constants

File: src/main/java/org/mskcc/utils/ParserUtil.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ParserUtil {
+    private static Logger log = LoggerFactory.getLogger(ParserUtil.class);
+
+    /**
+     * Safe conversion of a string to a float with logging
+     */
+    public static Double parseDouble(String st) {
```
Comment (dmcmanam) [inline src/main/java/org/mskcc/utils/ParserUtil.java]: Other options:
https://www.dariawan.com/tutorials/java/java-convert-string-double/

File: src/main/java/org/mskcc/crosscheckmetrics/model/CrosscheckMetrics.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+@Entity
+@ToString
+public class CrosscheckMetrics {
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    public Double lod;                // LOD score from picard CrosscheckFingerprints
+    public Double lodAlt1;
```
[the remark this answers is not in the document]
Comment (DavidStreid) [reply src/main/java/org/mskcc/crosscheckmetrics/model/CrosscheckMetrics.java]: sounds good

File: src/main/java/org/mskcc/constants.java (hunk 1 above)
Comment (DavidStreid) [reply src/main/java/org/mskcc/constants.java]: Definitely. My mistake

File: src/main/java/org/mskcc/utils/ParserUtil.java (hunk 2 above)
Comment (DavidStreid) [reply src/main/java/org/mskcc/utils/ParserUtil.java]: This wraps around that method, but I want to log the "java.lang.NumberFormatException" if it happens. Will try to add a splunk alert to catch this error in the logs and send an email

**Pull request merged**